processor/src/v1/decoder/mod.rs

```diff
@@ -0,0 +1,48 @@
+use vm_core::v1::{
+    program::{
+        blocks::{Join, Loop, OpBatch, Span, Split},
+        Operation,
+    },
+    BaseElement,
+};
+
+pub struct Decoder {}
+
+impl Decoder {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    // JOIN BLOCK
+    // --------------------------------------------------------------------------------------------
+
+    pub fn start_join(&mut self, _block: &Join) {}
```

Comment:
Does this self reference as well as the others need to be mutable?

Reply:
Yes. The reason is that we need to record the execution trace of the decoder and that will require updating internal structure (which is currently absent right now). But in general this will be similar to the current implementation of the stack (where there is `trace` member which gets updated with every cycle).

Reply:
Thanks for the clarification.

Reply:
No problem!